boto/utils.py

```diff
@@ -860,11 +864,12 @@ def notify(subject, body=None, html_body=None, to_string=None,
 
 
 def get_utf8_value(value):
-    if not six.PY2 and isinstance(value, bytes):
+    if six.PY3:
+        if isinstance(value, bytes):
+            return value.decode('utf-8')
         return value
-
     if not isinstance(value, six.string_types):
-        value = six.text_type(value)
+        value = six.text_type(value).encode('utf-8')
```

Comment:
This isn't necessary.  The next lines will detect that this is an instance of `text_type` and do the same encoding that's being done here.  This line can be reverted.

Reply:
Sounds right to me, reverted.